Core/src/main/java/org/tribuo/ensemble/WeightedEnsembleModel.java

```diff
@@ -216,4 +224,89 @@ public WeightedEnsembleModel(String name, EnsembleModelProvenance provenance, Im
+
+        // Build model
+        OnnxMl.ModelProto.Builder builder = OnnxMl.ModelProto.newBuilder();
+        builder.setGraph(context.buildGraph());
+        builder.setDomain(domain);
+        builder.setProducerName("Tribuo");
+        builder.setProducerVersion(Tribuo.VERSION);
+        builder.setModelVersion(modelVersion);
+        builder.setDocString(toString());
+        builder.addOpsetImport(ONNXOperators.getOpsetProto());
+        builder.setIrVersion(6);
+
+        // Extract provenance and store in metadata
+        OnnxMl.StringStringEntryProto.Builder metaBuilder = OnnxMl.StringStringEntryProto.newBuilder();
+        metaBuilder.setKey(ONNXExportable.PROVENANCE_METADATA_FIELD);
+        metaBuilder.setValue(serializeProvenance(getProvenance()));
+        builder.addMetadataProps(metaBuilder.build());
+
+        return builder.build();
+    }
+
+    @Override
+    public void writeONNXGraph(ONNXContext context, String inputName, String outputName) {
+        String[] memberOutputs = new String[models.size()];
+        // Write in each ensemble member
+        for (int i = 0; i < models.size(); i++) {
+            Model<T> model = models.get(i);
+            if (model instanceof ONNXExportable) {
+                memberOutputs[i] = context.generateUniqueName("ensemble_member_output");
+                ((ONNXExportable)model).writeONNXGraph(context,inputName,memberOutputs[i]);
+            } else {
+                throw new IllegalStateException("Ensemble member '" + model.toString() + "' is not ONNXExportable.");
+            }
+        }
+
+        // Unsqueeze the members so they have an extra dimension to concatenate along
+        OnnxMl.TensorProto unsqueezeAxes = ONNXUtils.arrayBuilder(context,"unsqueeze_ensemble_output",new long[]{2});
+        context.addInitializer(unsqueezeAxes);
+        String[] unsqueezeOutputs = new String[memberOutputs.length];
+        for (int i = 0; i < unsqueezeOutputs.length; i++) {
+            unsqueezeOutputs[i] = context.generateUniqueName("unsqueeze_output");
+            OnnxMl.NodeProto unsqueeze = ONNXOperators.UNSQUEEZE.build(context,new String[]{memberOutputs[i],unsqueezeAxes.getName()},unsqueezeOutputs[i]);
+            context.addNode(unsqueeze);
+        }
+
+        // Concatenate all the outputs into a single three tensor [batch_size, num_outputs, num_members]
+        OnnxMl.NodeProto concat = ONNXOperators.CONCAT.build(context,unsqueezeOutputs,context.generateUniqueName("ensemble_concat"), Collections.singletonMap("axis",2));
+        context.addNode(concat);
+
+        // Write in weights initializer
+        OnnxMl.TensorProto weightTensor = ONNXUtils.arrayBuilder(context,"ensemble_weights",weights);
+        context.addInitializer(weightTensor);
+
+        // Write in ensemble combiner
+        List<OnnxMl.NodeProto> combinerNodes = combiner.exportCombiner(context,concat.getOutput(0),outputName,weightTensor.getName());
+        if (combinerNodes.isEmpty()) {
```

Comment:
This is sort of an ugly hack, as that error message could potentially hide a different (also presumably erroneous) reason that the combiner is empty.

Reply:
The default method on `EnsembleCombiner` returns an empty list, and it's the same idiom as we used in the `VectorNormalizer`s in `LinearSGDModel`. I can update the docs for `exportCombiner` to say it only returns an empty list if the method isn't implemented, otherwise it should return the identity operation. And then say other erroneous cases are handled by throwing a runtime exception.